dotnet/src/Connectors/Connectors.Amazon/Bedrock/Models/Amazon/AmazonIOService.cs

```diff
@@ -0,0 +1,213 @@
+        };
+        return converseRequest;
+    }
+
+    private TValue GetExtensionDataValue<TValue>(IDictionary<string, object>? extensionData, string key, TValue defaultValue)
+    {
+        if (extensionData == null || !extensionData.TryGetValue(key, out object? value))
+        {
+            return defaultValue;
+        }
+
+        if (value is TValue typedValue)
+        {
+            return typedValue;
+        }
+
+        return defaultValue;
+    }
+
+    private static ConversationRole MapRole(AuthorRole role)
+    {
+        string roleStr;
+        if (role == AuthorRole.User)
+        {
+            roleStr = "user";
+        }
+        else
+        {
+            roleStr = "assistant";
+        }
+        return roleStr switch
+        {
+            "user" => ConversationRole.User,
+            "assistant" => ConversationRole.Assistant,
+            _ => throw new ArgumentOutOfRangeException(nameof(role), $"Invalid role: {role}")
+        };
+    }
+    /// <summary>
+    /// Extracts the text generation streaming output from the Amazon Titan response object structure.
+    /// </summary>
+    /// <param name="chunk"></param>
+    /// <returns></returns>
+    public IEnumerable<string> GetTextStreamOutput(JsonNode chunk)
+    {
+        var text = chunk["outputText"]?.ToString();
+        if (!string.IsNullOrEmpty(text))
+        {
+            yield return text;
+        }
+    }
+    /// <summary>
+    /// Builds the ConverseStreamRequest object for the Converse Bedrock API call, including building the Amazon Titan Request object and mapping parameters to the ConverseStreamRequest object.
+    /// </summary>
+    /// <param name="modelId">The model ID.</param>
+    /// <param name="chatHistory">The messages between assistant and user.</param>
+    /// <param name="settings">Optional prompt execution settings.</param>
+    /// <returns></returns>
+    public ConverseStreamRequest GetConverseStreamRequest(string modelId, ChatHistory chatHistory, PromptExecutionSettings settings)
+    {
+        var titanRequest = new TitanRequest.TitanChatCompletionRequest
+        {
+            Messages = chatHistory.Select(m => new Message
+            {
+                Role = MapRole(m.Role),
+                Content = new List<ContentBlock> { new() { Text = m.Content } }
+            }).ToList(),
+            System = new List<SystemContentBlock>(), // { new SystemContentBlock { Text = "You are an AI assistant." } },
+            InferenceConfig = new InferenceConfiguration
+            {
+                Temperature = this.GetExtensionDataValue(settings.ExtensionData, "temperature", 0.7f),
+                TopP = this.GetExtensionDataValue(settings.ExtensionData, "topP", 0.9f),
+                MaxTokens = this.GetExtensionDataValue(settings.ExtensionData, "maxTokenCount", 512),
+            },
+            AdditionalModelRequestFields = new Document(),
+            AdditionalModelResponseFieldPaths = new List<string>()
+        };
```

Comment:
Is there a need to convert inputs to `TitanRequest.TitanChatCompletionRequest` model here? Looks like only `ConverseStreamRequest` is needed

Reply:
Good note, I've removed all intermediary objects for getting the request for invoke, converse, and converse stream for all models.